doc/classes/VisualShaderNodeCustom.xml

```diff
@@ -0,0 +1,138 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<class name="VisualShaderNodeCustom" inherits="VisualShaderNode" category="Core" version="3.2">
+	<brief_description>
+		A custom visual shader node.
+	</brief_description>
+	<description>
+		By derivive from this class you can create a custom visual shader script addon which will automatically been added to visual shader editor. By overriding the provided virtual functions you can setup its behavior.
+		In order to correct work you must add a [code]tool[/code] keyword and provide a [code]class_name[/code] for your custom script.
+	</description>
+	<tutorials>
+	</tutorials>
+	<methods>
+		<method name="_get_category" qualifiers="virtual">
+			<return type="String">
+			</return>
+			<description>
+				By overriding this function, category of the associated custom node in the members dialog will be changed to the returned value.
+				Does not required to be overrided. If its not overrided, the "Custom" category will be used.
+			</description>
+			</method>
+		<method name="_get_description" qualifiers="virtual">
+			<return type="String">
+			</return>
+			<description>
+				By overriding this function, description of the associated custom node in the members dialog will be changed to the returned value.
+				Does not required to be overrided.
+			</description>
+		</method>
+		<method name="_get_code" qualifiers="virtual">
+			<return type="String">
+			</return>
+			<argument index="0" name="input_vars" type="Array">
+			</argument>
+			<argument index="1" name="output_vars" type="Array">
+			</argument>
+			<argument index="2" name="mode" type="int" enum="Shader.Mode">
+			</argument>	
+			<argument index="3" name="type" type="int" enum="VisualShader.Type">
+			</argument>
+			<description>
+				By overriding this function, the returned string will adds the actual shader code. You can modify the output ports values like: [code]return output_vars[0] + " = " + input_vars[0] + ";"[/code]. You can override the generated behavior dependens on the shader [code]mode[/code] (see [constant Shader.Mode]) and/or [code]type[/code] (see [constant VisualShader.Type]).
+				Required to be overrided.
+			</description>
+		</method>
+		<method name="_get_global_code" qualifiers="virtual">
+			<return type="String">
+			</return>
+			<argument index="0" name="mode" type="int" enum="Shader.Mode">
+			</argument>
+			<description>
+				By overriding this function, the returned string will adds the global code on the top of shader. You can define multiple function declarations, varyings, constants, uniforms etc. Be careful with this functionality because it may conflict with other custom nodes (give the defined entities the unique names). You can override the generated behavior dependens on the shader [code]mode[/code] (see [constant Shader.Mode]).
```

Comment:
Proofreading:
```
Override this method to add shader code on top of the global shader, to define your own standard library of reusable methods, varyings, constants, uniforms, etc. The shader code should be returned as a string, which can have multiple lines (the [code]"""[/code] multiline string construct can be used for convenience).
Be careful with this functionality as it can cause name conflicts with other custom nodes, so be sure to give the defined entities unique names.
You can customize the generated code based on the shader [code]mode[/code] (see [enum Shader.Mode]).
Defining this method is [b]optional[/b].
```